### PR TITLE
Redspace Thermite, thermite for traitors

### PR DESCRIPTION
**Pull request by neeshacark**
# Document the changes in your pull request

i don't know what im doing lol

thermite but it can melt rwalls, submitted as a draft so molti can help me work out how tf to code

Recipe is 1u thermite, 1u redspace dust (catalyst)

# Why is this good for the game?
gives traitors more options for breaking into places other than just emagging every door

# Testing

https://github.com/yogstation13/Yogstation/assets/44718209/15245fb4-7191-483a-80d8-4efa1f47739d

# Wiki Documentation

will need listing on the guide to chemistry

# Changelog

:cl: molti, cark
rscadd: redspace thermite, a thermite developed by traitors using redspace dust
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 1]
```diff
@@ -16,6 +16,24 @@
...
+/datum/reagent/thermite/redspace
+	name = "Redspace Thermite"
+	description = "Redspace Thermite produces an aluminothermic reaction known as a thermite reaction. Using the power of redspace dust, it is stronger than normal. Can be used to melt walls."
+	reagent_state = SOLID
+	color = "#550000"
+	taste_description = "sweet tasting metal and bitter evil"
+	process_flags = ORGANIC | SYNTHETIC
+	can_synth = FALSE
+
+/datum/reagent/thermite/reaction_turf(turf/T, reac_volume)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm]:
```suggestion
/datum/reagent/thermite/redspace/reaction_turf(turf/T, reac_volume)
```

File: code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 2]
```diff
@@ -16,6 +16,24 @@
...
+	if(reac_volume >= 1)
+		T.AddComponent(/datum/component/thermite, reac_volume)
+
+/datum/reagent/thermite/on_mob_life(mob/living/carbon/M)
+	M.adjustFireLoss(1, 0)
+	..()
+	return TRUE
+
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm]: delete this entirely, it'll just take the functionality from regular thermite so there's no need to change this

File: code/modules/reagents/chemistry/recipes/pyrotechnics.dm [hunk 3]
```diff
@@ -112,6 +112,13 @@
 	results = list(/datum/reagent/thermite = 3)
 	required_reagents = list(/datum/reagent/aluminium = 1, /datum/reagent/iron = 1, /datum/reagent/oxygen = 1)
 
+/datum/chemical_reaction/redspace_thermite
+	name = "Redspace Thermite"
+	id = /datum/reagent/redspace_thermite
+	results = list(/datum/reagent/redspace_thermite = 1)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/recipes/pyrotechnics.dm]:
```suggestion
	results = list(/datum/reagent/thermite/redspace = 1)
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm [hunk 4]
```diff
@@ -16,6 +16,19 @@
...
+	description = "Redspace Thermite produces an aluminothermic reaction known as a thermite reaction. Using the power of redspace dust, it is stronger than normal. Can be used to melt walls."
+	reagent_state = SOLID
+	color = "#550000"
+	taste_description = "sweet tasting metal and bitter evil"
+	process_flags = ORGANIC | SYNTHETIC
+	can_synth = FALSE
+
+/datum/reagent/thermite/redspace/reaction_turf(turf/T, reac_volume)
+	if(reac_volume >= 1)
+		T.AddComponent(/datum/component/thermite, reac_volume)
```
Comment (Moltijoe) [inline code/modules/reagents/chemistry/reagents/pyrotechnic_reagents.dm]:
```suggestion
		T.AddComponent(/datum/component/thermite, reac_volume, TRUE)
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Comment by neeshacark** [comment]
i am the worlds most intellectual coder

File: code/datums/components/thermite.dm [hunk 5]
```diff
@@ -13,17 +13,20 @@
...
 		/turf/closed/wall/mineral
 		)
 
-/datum/component/thermite/Initialize(_amount)
+/datum/component/thermite/Initialize(_amount, powerful = FALSE)
 	if(!istype(parent, /turf) || blacklist[parent.type])
 		return COMPONENT_INCOMPATIBLE
+
+	if(!powerful)
+		immunelist |= typecacheof(/turf/closed/wall/r_wall)
```
Comment (Moltijoe) [inline code/datums/components/thermite.dm]:
```suggestion
	if(powerful)
		resistlist |= typecacheof(/turf/closed/wall/r_wall)
	else
		immunelist |= typecacheof(/turf/closed/wall/r_wall)
```

Comment (Moltijoe) [reply code/datums/components/thermite.dm]: this is an idea that could be used for balance if needed

**Comment by Therandomhoboo** [comment]
So my gripes about making stuff specifically for antags with TC (because who cares about BB's) aside I don't massively dislike it however, my concern is that for using up actual TC it feels a bit lackluster. I can buy an emag or just hack the doors open quicker and without having to wait for the wall to come down which can get myself caught. 

I think the red thermite should act faster than normal thermite because of the effects red dust does in general (increases click speed and what not for people who huff it). I think this will make Red Thermite a bit more viable to use.

**Comment by Marmio64** [comment]
> So my gripes about making stuff specifically for antags with TC (because who cares about BB's) aside I don't massively dislike it however, my concern is that for using up actual TC it feels a bit lackluster. I can buy an emag or just hack the doors open quicker and without having to wait for the wall to come down which can get myself caught.
> 
> I think the red thermite should act faster than normal thermite because of the effects red dust does in general (increases click speed and what not for people who huff it). I think this will make Red Thermite a bit more viable to use.

the redspace dust is listed as a catalyst so it only takes 1 TC to turn however much thermite you want into red thermite. Doesn't need to be any stronger, especially because after you're done making however much red thermite you want you can also extract the redspace dust to use for something else.

**Comment by solwashere** [comment]
Redspace dust in lore: has properties that allows for the alteration of chemical bonds to be able to produce alternative means of already existing recipes so that agents in the field may use it to procure poisons not previously possible to make on NT made chem machines.
Redspace dust here: lol this makes explosion, funny.
I'd rather normal thermite be reworked anyways to allow for the destruction of rwalls but it is harder to make safely and takes forever to do so, longer than it'd take to just blow up a wall or deconstruct it.

**Comment by neeshacark** [comment]
> Redspace dust in lore: has properties that allows for the alteration of chemical bonds to be able to produce alternative means of already existing recipes so that agents in the field may use it to procure poisons not previously possible to make on NT made chem machines. Redspace dust here: lol this makes explosion, funny. I'd rather normal thermite be reworked anyways to allow for the destruction of rwalls but it is harder to make safely and takes forever to do so, longer than it'd take to just blow up a wall or deconstruct it.

we've been using redspace dust for traitor chemistry for some time now, so...

**Comment by Therandomhoboo** [comment]
> Redspace dust in lore: has properties that allows for the alteration of chemical bonds to be able to produce alternative means of already existing recipes so that agents in the field may use it to procure poisons not previously possible to make on NT made chem machines. Redspace dust here: lol this makes explosion, funny. I'd rather normal thermite be reworked anyways to allow for the destruction of rwalls but it is harder to make safely and takes forever to do so, longer than it'd take to just blow up a wall or deconstruct it.

"Gateway to traitor chemistry, want a drug to be traitor only? use this" How AZZZZ implemented this it was mainly for anything chemistry related. 

If I'm reading what you're saying correctly then this still alters the "chemical bonds" to become different so that it can destroy stronger material so it still makes sense lore wise. If you mean that redspace is only used to procure poisons then either red-eye chemical needs reworking (because it gives you a ton of buffs) or the lore does.

**Comment by SomeguyManperson** [comment]
Why should I buy c-4 instead of using this, since it seems like easily (and significantly cheaper) producable c-4 variant usable on _slightly_ less important stuff

**Comment by neeshacark** [comment]
Honestly, this PR has been so much more trouble that it was worth.  I did it mostly to dip my toes into coding a bit and it turned into this massive situation with people complaining over stuff and having inappropriate expectations for what people should do when making PRs that can in anyway be considered to 'not make sense within lore' (any pr that adds new chems/items i guess since this PR is somehow an issue)

There will be balance changes requested of me for this PR (fair enough) but I really don't care enough to apply them when the result will still be complaints over how this doesn't make sense in lore.  If this is what happens when a new feature is created that doesn't make sense to someone's perception of how this fictional world with literal wizards works - then I'm not going to bother with coding and will happily stick to mapping where there is literally almost zero way for lore to be infringed upon

**Pull request closed without merging**